src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,74 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final byte OPERATION_TYPE_INDEX = 0;
+    private static final byte AMMOUNT_INDEX = 1;
+    private static final byte SUPPLY_INDEX = 0;
+    private static final byte BUY_INDEX = 1;
+    private static final String SEPARATOR = ",";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] processedData = readFile(fromFileName);
+        writeFile(toFileName, getResultToWrite(processedData));
+    }
+
+    private int[] readFile(String fileName) {
+        String readString;
+        int[] totalSupplyAndBuy = new int[2];
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            readString = bufferedReader.readLine();
+            while (readString != null) {
+                totalSupplyAndBuy = dataProcessing(readString, totalSupplyAndBuy);
+                readString = bufferedReader.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File " + fileName + " not found", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading from file" + fileName, e);
+        }
+        return totalSupplyAndBuy;
+    }
+
+    private void writeFile(String fileName, String dataToWrite) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(fileName))) {
+            bufferedWriter.write(dataToWrite);
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing file " + fileName, e);
+        }
+    }
+
+    private int[] dataProcessing(String data, int[] totalSupplyAndBuy) {
```

Comment:
```suggestion
    private int[] calculate(String data) {
```
create the second variable inside of this method directly